nolearn/lasagne/base.py

```diff
@@ -481,6 +485,14 @@ def _create_iter_funcs(self, layers, objective, update, output_type):
         else:
             accuracy = loss_eval
 
+        scores_train, scores_valid = [], []
```

Comment:
You can remove the init `scores_train, scores_valid = [], []` and `if` check since you have made sure that `scores_train` and `scores_valid` are set to empty list.